app/controllers/attachment_masks_controller.rb

```diff
@@ -11,13 +11,29 @@ def wait
     file = @attachment.masked_file
 
     if file.attached?
-      redirect_to params[:referer]
+      redirect_to done_attachment_mask_path(
+        id: @attachment.to_signed_global_id,
+        referer: params[:referer]
+      )
 
     else
       FoiAttachmentMaskJob.perform_later(@attachment)
     end
   end
 
+  def done
```

Comment:
Looking at the commit message:

> Once an attachment masked file is available … redirecting to the download.

Redirecting to the actual attachment seems more preferable than requiring an extra click to do so…

> This is so we aren't left on the `wait` view which looks like nothing has happened.

…but I'm not sure I understand the conditions where this happens.

I'd have thought we could do:

```rb
def wait
  if @attachment.masked_file.attached?
    # If we have a cached file, render it
    redirect_to attachment_path(@attachment)
  else
    # otherwise, fire off a job to cache it, render the waiting page, and
    # keep refreshing until `@attachment.masked_file.attached?` returns
    # true
    FoiAttachmentMaskJob.perform_later(@attachment)
    render 'waiting'
  end
end
```

One to demo perhaps.

Reply:
After the `redirect_to attachment_path(@attachment)` the users browser would show the previous view as this will be treated link any other download link.

We could automatically download via JS or meta tag on the `done` view (and change the wording to say "Your download will start automatically...").

Happy to demo